Fix query string parameters when arg_separator.output is configured as &amp;

earlseay opened the pull request:
without html decoding the url here, the url get's messed up and doesn't honor any of the filter parameters past the first 1

earlseay committed:
html decode the url before you try to use it

without html decoding the url here, the url get's messed up and doesn't honor any of the filter parameters past the first 1

dereuromark commented:
ping @claudiosanches and team.
Sounds like a bug then that needs clarification.

joshuatf commented:
@earlseay Can you provide testing instructions for this PR?

michaelmeneses commented:
Hi,

When will it be released?

kraftbj committed:
Use explicit separator in http_build_query instead of html_entity_decode

This fixes the root cause by explicitly setting the '&' separator in
http_build_query(), which overrides the arg_separator.output PHP ini
setting. This is more precise than decoding the entire URL afterward.

michaelmeneses commented:
👏👏👏👏 

Great!

I've been waiting for this fix for a while.

Don't forget to update it on Packagist. 😃
